src/backend/distributed/deparser/deparse_foreign_server_stmts.c

```diff
@@ -0,0 +1,306 @@
+{
+	RenameStmt *stmt = castNode(RenameStmt, node);
+
+	Assert(stmt->renameType == OBJECT_FOREIGN_SERVER);
+
+	StringInfoData str;
+	initStringInfo(&str);
+
+	AppendAlterForeignServerRenameStmt(&str, stmt);
+
+	return str.data;
+}
+
+
+char *
+DeparseAlterForeignServerOwnerStmt(Node *node)
+{
+	AlterOwnerStmt *stmt = castNode(AlterOwnerStmt, node);
+
+	Assert(stmt->objectType == OBJECT_FOREIGN_SERVER);
+
+	StringInfoData str;
+	initStringInfo(&str);
+
+	AppendAlterForeignServerOwnerStmt(&str, stmt);
+
+	return str.data;
+}
+
+
+char *
+DeparseDropForeignServerStmt(Node *node)
+{
+	DropStmt *stmt = castNode(DropStmt, node);
+
+	Assert(stmt->removeType == OBJECT_FOREIGN_SERVER);
+
+	StringInfoData str;
+	initStringInfo(&str);
+
+	AppendDropForeignServerStmt(&str, stmt);
+
+	return str.data;
+}
+
+
+static void
+AppendCreateForeignServerStmt(StringInfo buf, CreateForeignServerStmt *stmt)
+{
+	appendStringInfoString(buf, "CREATE SERVER ");
+
+	if (stmt->if_not_exists)
+	{
+		appendStringInfoString(buf, "IF NOT EXISTS ");
+	}
+
+	appendStringInfo(buf, "%s ", quote_identifier(stmt->servername));
+
+	if (stmt->servertype)
+	{
+		appendStringInfo(buf, "TYPE %s ", quote_literal_cstr(stmt->servertype));
+	}
+
+	if (stmt->version)
+	{
+		appendStringInfo(buf, "VERSION %s ", quote_literal_cstr(stmt->version));
+	}
+
+	appendStringInfo(buf, "FOREIGN DATA WRAPPER %s ", quote_identifier(stmt->fdwname));
+
+	AppendCreateForeignServerOptions(buf, stmt);
+}
+
+
+static void
+AppendCreateForeignServerOptions(StringInfo buf, CreateForeignServerStmt *stmt)
+{
+	if (list_length(stmt->options) <= 0)
+	{
+		return;
+	}
+
+	appendStringInfoString(buf, "OPTIONS (");
+
+	DefElem *def = NULL;
+	foreach_ptr(def, stmt->options)
+	{
+		const char *value = quote_identifier(defGetString(def));
+		appendStringInfo(buf, "%s \'%s\'", def->defname, value);
+
+		if (def != llast(stmt->options))
+		{
+			appendStringInfoString(buf, ", ");
+		}
+	}
+
+	appendStringInfoString(buf, ")");
+}
+
+
+static void
+AppendAlterForeignServerStmt(StringInfo buf, AlterForeignServerStmt *stmt)
+{
+	appendStringInfo(buf, "ALTER SERVER %s ", quote_identifier(stmt->servername));
+
+	if (stmt->has_version)
+	{
+		appendStringInfo(buf, "VERSION %s ", stmt->version);
+	}
+
+	AppendAlterForeignServerOptions(buf, stmt);
+}
+
+
+static void
+AppendAlterForeignServerOptions(StringInfo buf, AlterForeignServerStmt *stmt)
+{
+	if (list_length(stmt->options) <= 0)
+	{
+		return;
+	}
+
+	appendStringInfoString(buf, "OPTIONS (");
+
+	DefElemAction action = DEFELEM_UNSPEC;
+	DefElem *def = NULL;
+	foreach_ptr(def, stmt->options)
+	{
+		if (def->defaction != DEFELEM_UNSPEC)
+		{
+			action = def->defaction;
+			char *actionString = GetDefElemActionString(action);
+			appendStringInfo(buf, "%s ", actionString);
+		}
+
+		appendStringInfo(buf, "%s", def->defname);
+
+		if (action != DEFELEM_DROP)
+		{
+			const char *value = defGetString(def);
+
+			appendStringInfo(buf, " \'%s\'", value);
```

Comment:
Any time you see quotes in an appendStringInfo, assume that there's a SQL injection bug.

Quoting should always be done separately to take into account escaping.

Reply:
I've added that for CREATE SERVER OPTIONS: https://github.com/citusdata/citus/pull/5468/files#diff-c2f60db4cb9c4046fb711bd8ded1841ad0cede72d4ed8239aaf91e6151e9f90eR150-R151

But apparently missed that for ALTER SERVER.

Could you please check the one that is for CREATE SERVER? I'm considering applying the same here